### DISCUSSION TITLE
Sharing netCDF data with WIS2box

### DISCUSSION THREAD

**Discussion by callumrollo** [wmo-im/wis2box#486] [Q&A]
Hi all,

I'm very new to sharing data via WIS2, so apologies if this has an obvious answer already.

I have a test WIS2box running and now I want to try sharing some of our own data. We have near real time oceanographic data from ocean gliders. Currently we produce timeseries-like netCDF files of pressure, temperature, salinity etc. https://observations.voiceoftheocean.org/

I have been struggling to get these datasets into WIS2. The demo data are all in the BUFR format, and I haven't yet found a library that can convert our netCDF data into the glider BUFR format Table D: Category 15 (oceanography), Table reference 3-15-012 that was added to wmo-if/BUFR4 earlier this year https://github.com/wmo-im/BUFR4/pull/146

I have not worked with BUFR before and I'm struggling to make valid BUFR files. Could someone point me toward a good resource for making this transformation? 

I remember reading that WIS2 will be able to take in data in formats other than BUFR. Is this currently the case? If I convert our timeseries data into a csv file, could I use https://github.com/wmo-im/csv2bufr to pull it into WIS2? Or is it best to make BUFR files?

Many thank
Callum

**Comment by david-i-berry** [comment]
Hi Callum,

Apologies for the slow reply, I've been working on other things but have recently returned my focus on to the WIS2 (and wis2box). I might be able to help with the conversion to BUFR, alternatively NetCDF is a valid WMO format. The WMO Manual on Codes (WMO-NO. 306, Volume I.2) is in the process of being updated to include NetCDF and two profiles based on the CF-Conventions, one for trajectory data and the other for CF-Radial (weather radar). I'm about to travel and be out of the office for the next 3 weeks but should be able to help more after then. Until then, the NetCDF profile can be found at: https://community.wmo.int/en/activity-areas/wis/wmo-cf-extensions and the WMO-CF Marine Trajectory. 

Best regards,

David Berry.

**Comment by callumrollo** [comment]
Thanks for the linK! I'll work on making our netcdf data compatible with the ocean profile standard.
